### ISSUE TITLE
FCN Crop Layer

### ISSUE THREAD

**Issue by ziky90** [tensorflow/tensorflow#2049]
Hi all,

I am currently trying to implement FCN for semantic segmentation in TensorFlow as it was previously done in Caffe here https://github.com/shelhamer/fcn.berkeleyvision.org.

Unfortunately I'm struggling with following 3 things:

1) How to map `Deconvolution` layer from Caffe to TensorFlow? But I guess that I have found the solution in `tf.nn.conv2d_transpose`?
2) How to map `Crop` layer from Caffe to TensorFlow? **Unfortunately I can't see any alternative in TensorFlow. Is there equivalent for this?**
3) Does Caffe `SoftmaxWithLoss` correspond to TensorFlow softmax_cross_entropy_with_logits?

Thank you in advance for any advices, hints and help.

**Comment by vrv** [comment]
I am not famiilar with Caffe but:

1. yes, conv2d_transpose is what a lot of people call 'Deconvolution': https://www.tensorflow.org/versions/r0.8/api_docs/python/nn.html#conv2d_transpose
2. Also not sure what Crop does (can't find documentation for it), but check here: https://www.tensorflow.org/versions/r0.8/api_docs/python/image.html#cropping
3. https://www.tensorflow.org/versions/r0.8/api_docs/python/nn.html#softmax_cross_entropy_with_logits is our documentation for the math behind the op -- I cannot find similar mathematical description of SoftmaxWithLoss here: http://caffe.berkeleyvision.org/tutorial/loss.html, so I can only assume that they are the same.

Lastly, these are questions better asked on StackOverflow, only people looking at bugs / issues are really looking here.

**Comment by vrv** [comment]
(closing -- further discussion should be on StackOverflow)

**Comment by ziky90** [comment]
@vrv Thank you very much for the information.
I have tried asking on [StackOverflow](http://stackoverflow.com/questions/36746860/fcn-in-tensorflow-missing-crop-layer) and [Google groups](https://groups.google.com/a/tensorflow.org/forum/#!topic/discuss/WC3tyFXILvE) before, but with no answer. This was the reason for creating the issue also here.
